src/handlers/artist.js

```diff
@@ -3,6 +3,27 @@ const fs = require('fs');
 
 
 const artistHandler = (res, endpoint) => {
-
-}
+  fs.readFile(path.join(__dirname, '../post.json'), (err, data) => {
+    if (err) {
+      res.writeHead(500, { 'Content-type': 'html/text' });
+      res.write('<h1> Internal Server Error ! <h1>');
+      res.end();
+    } 
+    else {
+      const value = endpoint.slice(endpoint.lastIndexOf('/') + 1);
+      let count = 0;
+      const parsedData = JSON.parse(data);
+      const filteredData =parsedData.songs.filter((ele) => {
+        if (ele.artist.toUpperCase().includes(value.toUpperCase()) && count < 5) {
+          // eslint-disable-next-line no-plusplus
+          count++;
+          return ele;
```

Comment:
I guess this return will end the function execution and so the lines where you send the response won't be reached and so the the request will be pending in this case.